### PR TITLE
rspec-rubocop autofix the specs

### PR DESCRIPTION
**Pull request by tas50**
I'm not really a fan of the described_class changes, but I'm opening this to start a convo of how we could potentially get some of these rules into chefstyle

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: spec/functional/application_spec.rb [hunk 1]
```diff
@@ -26,12 +26,12 @@
   let(:argv) { [] }
   let(:stderr) { StringIO.new }
 
-  before(:each) do
+  before do
```
Comment (lamont-granquist) [inline spec/functional/application_spec.rb]: i'm not sure i'm wild about this one, makes it easier for newbies to clearly see that `:each` is the default.

File: spec/functional/plugins/powershell_spec.rb [hunk 2]
```diff
@@ -22,10 +22,10 @@
   VERSION_MATCHING_REGEX = /^(?:[\d]+\.)+[\d]+$/
   describe "powershell plugin", :windows_only do
     RSpec.shared_examples "a version looking thing" do
-      it "should be present" do
+      it "is present" do
```
Comment (lamont-granquist) [inline spec/functional/plugins/powershell_spec.rb]: love these

File: spec/unit/application_spec.rb [hunk 3]
```diff
@@ -87,7 +87,7 @@
       let(:directory) { "/some/fantastic/plugins" }
 
       it "does not generate deprecated config warnings for cli options" do
-        expect(Ohai::Log).to_not receive(:warn).
+        expect(Ohai::Log).not_to receive(:warn).
```
Comment (lamont-granquist) [inline spec/unit/application_spec.rb]: this is also good

File: spec/unit/dsl/plugin_spec.rb [hunk 4]
```diff
@@ -282,12 +282,12 @@
     describe "a top-level attribute" do
       describe "as a string" do
         it "returns false when the attribute does not exist" do
-          expect(plugin.attribute?("alice in chains")).to eql(false)
+          expect(plugin.attribute?("alice in chains")).to be(false)
```
Comment (lamont-granquist) [inline spec/unit/dsl/plugin_spec.rb]: should really `be false` but it's a step in the right direction

File: spec/unit/plugins/dmi_spec.rb [hunk 5]
```diff
@@ -141,7 +141,7 @@
     end
   end
 
-  it "should correctly ignore unwanted data" do
+  it "correctlies ignore unwanted data" do
```
Comment (lamont-granquist) [inline spec/unit/plugins/dmi_spec.rb]: English is hard...

File: spec/unit/plugins/elixir_spec.rb [hunk 6]
```diff
@@ -19,11 +19,11 @@
 describe Ohai::System, "plugin elixir" do
   let(:plugin) { get_plugin("elixir") }
 
-  before(:each) do
+  before do
     plugin[:languages] = Mash.new
   end
 
-  it "should shellout to elixir -v" do
+  it "shellouts to elixir -v" do
```
Comment (lamont-granquist) [inline spec/unit/plugins/elixir_spec.rb]: reads awkward to me

File: spec/unit/plugins/hostname_spec.rb [hunk 7]
```diff
@@ -19,95 +19,95 @@
...
 
   context "when a system has a bare hostname without a FQDN" do
-    before(:each) do
+    before do
       allow(@plugin).to receive(:collect_os).and_return(:default)
       allow(@plugin).to receive(:shell_out).with("hostname").and_return(mock_shell_out(0, "katie", ""))
     end
 
-    it "should correctly set the [short] hostname" do
+    it "correctlies set the [short] hostname" do
```
Comment (lamont-granquist) [inline spec/unit/plugins/hostname_spec.rb]: lol English again

**Comment by lamont-granquist** [comment]
yeah dunno, it doesn't feel like this is actually offering much utility in making actually better specs.  the ones that attempt to fix the English actually wind up mangling it some cases.  the most useful one i see is the `not_to/to_not` one because it's annoyed me that i can't find a consistent answer on which one to prefer and i hate having the choice.   the `eql(true)` => `be(true)` one is the other somewhat useful one.

File: spec/unit/plugins/dmi_spec.rb (hunk 5 above)
Comment (tas50) [inline spec/unit/plugins/dmi_spec.rb]: It mangled a bunch of them. Clearly I missed this one

File: spec/unit/plugins/elixir_spec.rb (hunk 6 above)
Comment (tas50) [inline spec/unit/plugins/elixir_spec.rb]: yep

**Comment by tas50** [comment]
Closing this for now. Some day we might come up with a rule set that makes sense for all chef projects. Today is not that day

**Comment by lamont-granquist** [comment]
yeah i'm not sure that rspec-rubocop is all that mature right now

**Comment by tas50** [comment]
Some day

**Pull request closed without merging**